### PR TITLE
1296 put aws lambda code into source control

### LINKED ISSUE #1296: Put AWS lambda code into source control

### PR DESCRIPTION
**Pull request by sadatmalik**
This PR:

- Reorganises the infra/aws folder into 2 subfolders infra/aws/terrafom and infra/aws/lambdas
- Moves all terraform scripts and confg under terraform/
- Adds ecs redeployment lambda code under lambdas/
- Update terraform git actions

### CODE REVIEW

File: infra/aws/lambdas/ECSServiceRedployFucntion.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#  along with this program. If not, see https://www.gnu.org/licenses/.
+
+import boto3
+import json
+from datetime import datetime
+
+
+def lambda_handler(event, context):
+    # Initialize ECS client using Boto3 to interact with ECS service
+    ecs_client = boto3.client('ecs')
```
Comment (sadatmalik) [inline infra/aws/lambdas/ECSServiceRedployFucntion.py:22]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

A Boto3 client is initialized from a Lambda handler. To speed up your Boto3 client initialization and minimize the operational cost of your Lambda function, we recommend that you create the client at the level of the module that contains the handler, and then reuse it between invocations.

[Learn more](https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html#function-code)

Suggested remediation:
We recommended the following changes to mitigate the issue.
```
@@ -20,5 +21,5 @@
 def lambda_handler(event, context):
     # Initialize ECS client using Boto3 to interact with ECS service
-    ecs_client = boto3.client(&#39;ecs&#39;)
+    
 
     # Define the ECS cluster and service to be updated

```

Comment (sadatmalik) [reply infra/aws/lambdas/ECSServiceRedployFucntion.py:22]: We only call the boto3 client once so pre initialising doesn't save us any operational cost in the case of this lambda.

File: infra/aws/lambdas/ECSServiceRedployFucntion.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    # Define the ECS cluster and service to be updated
+    cluster_name = 'tctalent-test'
+    service_name = 'tctalent-test'
```
Comment (camerojo) [inline infra/aws/lambdas/ECSServiceRedployFucntion.py:26]: Maybe a note to change this when configuring it for production?

**Review by camerojo — APPROVED** [review]
Approved - but see comment @sadatmalik

File: infra/aws/lambdas/ECSServiceRedployFucntion.py (hunk 2 above)
Comment (sadatmalik) [reply infra/aws/lambdas/ECSServiceRedployFucntion.py:26]: I'll merge these changes and add a note to clarify the above in my next PR

**Pull request merged**